### PR TITLE
New trait: Unstable

### PR DESCRIPTION
**Pull request by tralezab**
:cl:
add: New negative trait: Unstable!
/:cl:

Your sanity can go down, but it can never go up. Be very careful with your sanity...

Having high moods is still important to avoid any unexpected bad times lowering your sanity. You don't want to end up an insane wreck by the end of the round, right? I like the idea that this perk turns sanity into a resource that cannot be gained back, and thus must be "spent" getting into risky situations throughout the round.

### CODE REVIEW

**Comment by necromanceranne** [comment]
Oof, losing max health is a rough one. I think at -1 that's a pretty harsh trait to take, especially since you need to be meticulous about what you do in a round not to dumpster your sanity. Some is just unavoidable until it you encounter it, like breathloss. Other stuff is innocuously deadly like the fucking clown chain slipping you or getting chain tabled.

**Comment by Shadowflame909** [comment]
I think this should be worth more


A clown with pies or hygiene will really ruin a round

**Comment by Repukan** [comment]
Why should it be worth more?

It's not mandatory to take. You either take it for RP or for the extra challenge. Not everything needs to be put on par for powergaming

**Comment by Mickyan** [comment]
FYI you can recover sanity with psicodine, that considering 1 point sounds alright

**Comment by tralezab** [comment]
It's 1 point because there is no downside roundstart, and it's a free point if you dont live through the shit that could make you insane. I'm probably going to remove recovering sanity with psicodine as a lot of quirks use chems as their gimmick

**Comment by necromanceranne** [comment]
Most of the positive quirks aren't really powergamey, they just get you some nice minor stuff. The only real powergamey one is nightvision if you know how to use it. 

But the ones which do have long term effect on the round tend to be worth more. While it does nothing roundstart, it will come up eventually and will impact the round in some way.

**Comment by Repukan** [comment]
Then don't take it

???????

**Comment by necromanceranne** [comment]
If I'm taking something that impacts my round potentially as hard as guaranteed insanity, even for RP reasons, I'd like at least some counterbalance as a net positive. I already only use the quirks for RP but since they have gameplay impact it's worth considering how much it gives back for taking it.

**Comment by tralezab** [comment]
I'll give it an extra point, but make insanity more dangerous to go along with it.

**Comment by anconfuzedrock** [comment]
I think “be sad for more than a minute and you’re slower than any combatant for the rest of the round as well as act slower” is probably bad enough, really.

File: code/datums/components/mood.dm [hunk 1]
```diff
@@ -212,7 +216,10 @@
 		amount = sanity + 0.5
 	else if(sanity > maximum && amount > sanity - 0.5)
 		amount = sanity - 0.5
-	sanity = amount
+
+	// Disturbed stops you from getting any more sane
+	if(!(owner.has_trait(TRAIT_UNSTABLE) && amount > sanity))
+		sanity = amount
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/components/mood.dm]: "If you don't have the sanity trait and your adjusted amount is higher than current sanity, set sanity to adjusted amount"

That doesn't sound right.

Really you could just remove this check and put it on the check before +0.5 addition

`if(!(owner.has_trait(TRAIT_UNSTABLE)) && sanity < minimum && amount < sanity + 0.5)`

**Comment by Therapist1** [comment]
Maybe make them start self harming at low sanity (punch themselves, hit themselves with whatever they’re holding, refuse to take meds by their own hand) and at critical sanity loss, tear out their eyes/ears/tongue.

File: code/datums/components/mood.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/components/mood.dm]: okay thanks, i'm never good with these

**Comment by tralezab** [comment]
confused rock does leave a good point though, i'll keep it visual to stay balanced

File: code/datums/components/mood.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/datums/components/mood.dm]: actually that might not work if you have an amount higher than sanity hmmmm

Comment (ExcessiveUseOfCobblestone) [reply code/datums/components/mood.dm]:
```DM
if(owner.has_trait(TRAIT_UNSTABLE))
    sanity = min(amount,sanity)
else
    sanity = amount
```

That might be more appropriate?

**Comment by Chronitonity** [comment]
Unstable sounds more like all moods would effect you more rather then just making it impossible to become happier but maybe thats just me

**Comment by anconfuzedrock** [comment]
thas what hypersensitive does

File: code/datums/components/mood.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/components/mood.dm]: ok that's perfect and very clear in what it does

**Comment by ExcessiveUseOfCobblestone** [comment]
lol u forgot to add the file

**Comment by tralezab** [comment]
Ready

**Pull request merged**